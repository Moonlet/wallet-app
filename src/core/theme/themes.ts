export enum Theme {
    dark = 'dark',
    light = 'light'
}
