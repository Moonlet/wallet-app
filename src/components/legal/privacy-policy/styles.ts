import { StyleSheet } from 'react-native';
import { ITheme } from '../../../core/theme/itheme';
import { BASE_DIMENSION, ICON_CONTAINER_SIZE, SCREEN_HEIGHT } from '../../../styles/dimensions';
import { normalize } from '../../../library';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            flex: 1,
            justifyContent: 'center',
            backgroundColor: theme.colors.appBackground
        },
        button: {
            flexDirection: 'row',
            alignItems: 'center'
        },
        iconContainer: {
            width: normalize(ICON_CONTAINER_SIZE),
            height: normalize(ICON_CONTAINER_SIZE),
            alignItems: 'center',
            justifyContent: 'center',
            marginLeft: normalize(BASE_DIMENSION)
        },
        icon: {
            color: theme.colors.accent
        },
        webviewContainer: {
            flex: 1,
            overflow: 'hidden',
            backgroundColor: theme.colors.appBackground,
            height: SCREEN_HEIGHT // used for web
        },
        loadingIndicator: {
            position: 'absolute',
            top: 0,
            bottom: 0,
            left: 0,
            right: 0,
            backgroundColor: theme.colors.appBackground
        }
    });
