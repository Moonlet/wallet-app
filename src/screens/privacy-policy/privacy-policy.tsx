import React from 'react';
import { View } from 'react-native';
import { NavigationParams, NavigationScreenProp, NavigationState } from 'react-navigation';

import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';
import { Text } from '../../library';
import { translate } from '../../core/i18n';

export interface IProps {
    navigation: NavigationScreenProp<NavigationState, NavigationParams>;
    styles: ReturnType<typeof stylesProvider>;
}

export const PrivacyPolicyScreenComponent = (props: IProps) => (
    <View style={props.styles.container}>
        <Text>Privacy policy scroll</Text>
    </View>
);

export const navigationOptions = () => ({
    title: translate('App.labels.privacyPolicy')
});

export const PrivacyPolicyScreen = withTheme(stylesProvider)(PrivacyPolicyScreenComponent);

PrivacyPolicyScreen.navigationOptions = navigationOptions;
