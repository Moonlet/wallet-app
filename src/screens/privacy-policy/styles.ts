import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            paddingHorizontal: 16,
            alignItems: 'center',
            justifyContent: 'space-between',
            backgroundColor: theme.colors.appBackground,
            ...StyleSheet.absoluteFillObject
        }
    });
