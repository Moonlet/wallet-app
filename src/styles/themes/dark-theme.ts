import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
