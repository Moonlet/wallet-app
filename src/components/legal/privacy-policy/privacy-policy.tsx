import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { withTheme, IThemeProps } from '../../../core/theme/with-theme';
import stylesProvider from './styles';
import { smartConnect } from '../../../core/utils/smart-connect';
import Icon from '../../icon';
import { ICON_SIZE } from '../../../styles/dimensions';
import { LoadingIndicator } from '../../loading-indicator/loading-indicator';
import { WebView } from '../../../library/webview/webview';
import { SafeAreaView } from 'react-navigation';
import { CONFIG } from '../../../config';

interface IExternalProps {
    onClose?: () => void;
    showClose?: boolean;
}

export const PrivacyPolicyComponent = (
    props: IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>
) => {
    return (
        <SafeAreaView forceInset={{ bottom: 'never' }} style={props.styles.container}>
            <React.Fragment>
                {props.showClose && (
                    <TouchableOpacity
                        onPress={() => props.onClose && props.onClose()}
                        style={props.styles.button}
                    >
                        <View style={props.styles.iconContainer}>
                            <Icon name="close" size={ICON_SIZE} style={props.styles.icon} />
                        </View>
                    </TouchableOpacity>
                )}
                <View style={props.styles.webviewContainer}>
                    <WebView
                        startInLoadingState={true}
                        source={{ uri: CONFIG.privacyPolicyUrl }}
                        renderLoading={() => (
                            <View style={props.styles.loadingIndicator}>
                                <LoadingIndicator />
                            </View>
                        )}
                    />
                </View>
            </React.Fragment>
        </SafeAreaView>
    );
};

export const PrivacyPolicy = smartConnect<IExternalProps>(PrivacyPolicyComponent, [
    withTheme(stylesProvider)
]);
