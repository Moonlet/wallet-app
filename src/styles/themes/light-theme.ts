import { ITheme } from '../../core/theme/itheme';
import { darkTheme } from './dark-theme';

export const lightTheme: ITheme = darkTheme;
