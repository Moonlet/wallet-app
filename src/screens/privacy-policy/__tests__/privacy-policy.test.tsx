import React from 'react';
import { PrivacyPolicyScreenComponent, IProps } from '../privacy-policy';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';

const props: IProps = {
    // @ts-ignore
    navigation: {
        navigate: jest.fn()
    },
    styles: styleProvider(darkTheme)
};

describe('privacy policy screen component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<PrivacyPolicyScreenComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
