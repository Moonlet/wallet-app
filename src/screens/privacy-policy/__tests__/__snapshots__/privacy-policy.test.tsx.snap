// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`privacy policy screen component renders correctly 1`] = `
"<View style={{...}}>
  <withTheme(TextComponent)>
    Privacy policy scroll
  </withTheme(TextComponent)>
</View>"
`;
